doc/tensor.md

```diff
@@ -1327,36 +1331,66 @@ x:indexFill(2,torch.LongTensor{4,2},-10)
 ```
 
 <a name="torch.Tensor.gather"/>
-### [Tensor] gather(src, dim, index) ###
+### [Tensor] gather(dim, index) ###
```

Comment:
If I understand correctly, it's true for many of the functions that you can call them either as `src:func()` or `torch.func(src)`. Is the point here that this function can additionally be called as `dst:func(src)` which doesn't work for all functions? If that's the case then it's perhaps best to present the `src:func()` case in the heading, which would be consistent with the other functions, and familiar to the reader, and then make an explicit note in the description that it can be called the `dst:func(src)` way? Perhaps the way it is done at the bottom of the description of `index` "It is possible to store the result into an existing Tensor with `result:index(source, ...)`"

How does that sound?